File: client/src/app/_models/expense.ts
export interface IExpense {
  id?: number
  amount: number;
  month: string;
  category: string;
  year: number;
  type: string;
}

export interface IGroupedExpense {
  year: number
  month: string
  expenses: IExpenseInGrouped[]
}

export interface IExpenseInGrouped {
  id: number
  amount: number
  category: string
  type: string
}

File: API/Models/Expense.cs
namespace API.Models;

public class Expense
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public string Month { get; set; }
    public string Category { get; set; }
    public int Year { get; set; }
    public string Type { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow; 
}